mlflow/store/sqlalchemy_store.py

```diff
@@ -463,6 +463,26 @@ def set_tag(self, run_id, tag):
             self._check_run_is_active(run)
             session.merge(SqlTag(run_uuid=run_id, key=tag.key, value=tag.value))
 
+    def delete_tag(self, run_id, key):
+        """
+        Delete a tag from a run.
+        :param run_id: ID of the run that contains the tag to delete
+        :param key: name of the tag to delete
+        """
+        with self.ManagedSessionMaker() as session:
+            run = self._get_run(run_uuid=run_id, session=session)
+            self._check_run_is_active(run)
+            filtered_tags = session.query(SqlTag).filter_by(key=key).all()
```

Comment:
Looks like this `filter_by` clause still doesn't specify `run_id` (though this change was pushed to https://github.com/mlflow/mlflow/pull/1559).